lib/errors.js

```diff
@@ -0,0 +1,33 @@
+'use babel';
+
+//
+// Utility functions for parsing errors
+//
+
+const level2severity = (level) => {
+  switch (level) {
+    case 'warning': return 'warning';
+    case 'error': return 'error';
+    case 'note': return 'info';
+    case 'help': return 'info';
+    default: return 'error';
+  }
+};
+
+const level2type = (level) => {
+  return level.charAt(0).toUpperCase() + level.slice(1);
+};
+
+// Set location for special cases when the compiler doesn't provide it
+function preprocessMessage(msg, buildWorkDir) {
+  if (msg.file) {
+    return true;
+  }
+  if (msg.message === 'main function not found') {
+    msg.file = buildWorkDir + '/src/main.rs';	// TODO: When running an example, set the path to the example file.
```

Comment:
I tried pointing it to `Cargo.toml` but then switched back because it's very unlikely that `Cargo.toml` is the file in which the developer will be fixing this problem. It's usually either `src/main.rs` or one of the examples. Making difference between `main.rs` and an example file requires passing a wider building context to this function. Since we don't do it now, I just left `main.rs`.

I'm not sure what solution is better. `Cargo.toml` is universal, but seems useless from the practical point of view.

Regarding filing a bug, I'm going to do that after studying it a bit closer to be able to come up with some practical suggestions. I [told Jonathan](https://internals.rust-lang.org/t/editor-compatibility-and-the-new-error-format/3435/32?u=alygin) about it but to no avail so far.